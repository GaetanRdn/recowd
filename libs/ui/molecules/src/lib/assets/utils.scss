@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
