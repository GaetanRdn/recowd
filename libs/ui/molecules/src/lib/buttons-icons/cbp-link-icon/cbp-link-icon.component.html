<a href="https://www.cbp.fr/" rcButtonIcon target="_blank">
  <rc-cbp-custom-icon></rc-cbp-custom-icon>
</a>
