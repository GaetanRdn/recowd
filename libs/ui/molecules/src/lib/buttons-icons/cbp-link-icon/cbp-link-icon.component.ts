import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonDirective } from '@recowd/ui-atoms';
import { CbpCustomIconComponent } from '../../custom-icons/cbp-custom-icon/cbp-custom-icon.component';

@Component({
  selector: 'rc-cbp-link-icon',
  standalone: true,
  imports: [ButtonDirective, CbpCustomIconComponent],
  templateUrl: './cbp-link-icon.component.html',
  styleUrls: ['./cbp-link-icon.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CbpLinkIconComponent {}
