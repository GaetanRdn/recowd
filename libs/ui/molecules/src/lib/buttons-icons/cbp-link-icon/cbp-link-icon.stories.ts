import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { CbpLinkIconComponent } from './cbp-link-icon.component';

export default {
  title: 'molecules/buttons-icons/cbp-link-icon',
  parameters: {
    jest: 'cbp-link-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [CbpLinkIconComponent],
    }),
  ],
} as Meta<CbpLinkIconComponent>;

const Template: Story<CbpLinkIconComponent> = () => ({
  template: `<rc-cbp-link-icon></rc-cbp-link-icon>`,
});

export const Default = Template.bind({});
