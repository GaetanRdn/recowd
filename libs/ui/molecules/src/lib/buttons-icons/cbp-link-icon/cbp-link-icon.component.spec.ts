import { TestBed, waitForAsync } from '@angular/core/testing';

import { CbpLinkIconComponent } from './cbp-link-icon.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('CbpLinkIconComponent', () => {
  let templateLookup: TemplateLookup<CbpLinkIconComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [CbpLinkIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(CbpLinkIconComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
