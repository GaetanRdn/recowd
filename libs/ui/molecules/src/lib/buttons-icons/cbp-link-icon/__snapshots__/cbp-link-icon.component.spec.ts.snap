// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CbpLinkIconComponent should create 1`] = `
HTMLCollection [
  <a
    class="rc-button-icon"
    href="https://www.cbp.fr/"
    rcbuttonicon=""
    target="_blank"
  >
    <rc-cbp-custom-icon>
      <rc-custom-icon>
        <object
          data="assets/images/cbp.svg"
          height="300"
          type="image/svg+xml"
          width="300"
        >
          <img
            alt="assets/images/cbp"
            src="assets/images/cbp.png"
          />
        </object>
      </rc-custom-icon>
    </rc-cbp-custom-icon>
  </a>,
]
`;
